internal/ethapi/api.go

```diff
@@ -1427,10 +1427,18 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 	}
 	receipt := receipts[index]
 
+	txType := tx.Type()
 	var signer types.Signer = types.FrontierSigner{}
 	if tx.Protected() {
-		signer = types.NewEIP155Signer(tx.ChainId())
+		switch txType {
```

Comment:
This is not correct. We should use `types.MakeSigner` with the block number here. I'll fix it.

Reply:
Cool, I took a stab at a fix here: https://github.com/ryanschneider/go-ethereum/commit/1ce3baeb2f947dea48719d7a206f748574c471d5  @fjl feel free to cherry-pick in if it looks correct.  Note that if this fix is correct we probably need to do something similar in `newRPCTransaction`.

Reply:
I fixed it in an even-simpler way. Still thinking about the right solution for newRPCTransaction